src/arg_max_min.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './lib/tensor.js';
+import {reduceMax, reduceMin, selectValuesToReduce} from './reduce.js';
+import {squeeze} from './squeeze.js';
+
+/**
+ * Get the index location of the minimum or maxmium values of all the input values along the axes.
+ * @param {Tensor} input
+ * @param {Function} reduceFunc
+ * @param {MLArgMinMaxOptions} [options]
+ * @return {Tensor}
+ */
+export function argMaxMin(
+    input,
+    reduceFunc,
+    {
+      axes = null,
+      keepDimensions = false,
+      selectLastIndex = false,
+    } = {}) {
+  const inpAxes = axes ?? new Array(input.rank).fill(0).map((_, i) => i);
+  const outputShape = input.shape.slice();
+
+  for (let i = 0; i < inpAxes.length; ++i) {
+    outputShape[inpAxes[i]] = 1;
+  }
+
+  let output = new Tensor(outputShape);
+  const tensor = reduceFunc(input, {axes: inpAxes, keepDimensions: true});
+
+  for (let outputIndex = 0; outputIndex < sizeOfShape(outputShape); ++outputIndex) {
+    const value = tensor.getValueByIndex(outputIndex);
+    const inputLocation = output.locationFromIndex(outputIndex);
+    const selectedArray = selectValuesToReduce(input, inpAxes, inputLocation);
+    const index =
+        selectLastIndex ? selectedArray.lastIndexOf(value) : selectedArray.indexOf(value);
+    output.setValueByIndex(outputIndex, index);
+  }
+
+  if (!keepDimensions) {
+    output = squeeze(output);
```

Comment:
You may need to specify axes here, otherwise squeeze will delete all axes with the value of 1. such as:
a.shape =  {1, 2, 3, 4}, a.argMax(axes={1}, keepDimensions = false), then the output shape should be {1, 3, 4}.

Reply:
Good catch! Thanks @miaobin. 
It has been fixed, please take another look.

Reply:
Thanks for the fix and the addition of the corresponding test cases, LGTM.